### ISSUE TITLE
sequel: Object of type Lorg/graalvm/compiler/truffle/runtime/FrameWithoutBoxing;  should not be materialized (must not pass virtual object into an invoke that cannot be inlined)

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1630]
Unfortunately, I can't localize a test causing this issue (one of reason why I dislike minitest/test-unit in Ruby).

I guess the issue can be associated to `pg`

To reproduce:
```
git clone https://github.com/jeremyevans/sequel.git
cd sequel
bundle 
bundle exec rake spec_travis
```

Error:

```
~/.rubies/truffleruby-1.0.0-rc14/bin/truffleruby spec/core_spec.rb
Run options: --seed 48140

# Running:

........................................................................................................................................................................................................................................................................................................................................................................................................................................................E....................................................................................................................................................................................................................................................................................................................................F............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................org.graalvm.compiler.code.SourceStackTraceBailoutException$1: Object of type Lorg/graalvm/compiler/truffle/runtime/FrameWithoutBoxing; should not be materialized (must not pass virtual object into an invoke that cannot be inlined):
	at org.truffleruby.language.exceptions.RescueNode.execute(RescueNode.java:41)
	at org.truffleruby.language.exceptions.SetExceptionVariableNode.setLastExceptionAndRun(SetExceptionVariableNode.java:35)
	at org.truffleruby.language.exceptions.TryNode.setLastExceptionAndRunRescue(TryNode.java:107)
	at org.truffleruby.language.exceptions.TryNode.handleException(TryNode.java:93)
	at org.truffleruby.language.exceptions.TryNode.execute(TryNode.java:55)
	at org.truffleruby.language.control.SequenceNode.execute(SequenceNode.java:34)
	at org.truffleruby.language.methods.CatchForMethodNode.execute(CatchForMethodNode.java:38)
	at org.truffleruby.language.methods.ExceptionTranslatingNode.execute(ExceptionTranslatingNode.java:51)
	at org.truffleruby.language.RubyRootNode.execute(RubyRootNode.java:54)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callProxy(OptimizedCallTarget.java:274)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callRoot(OptimizedCallTarget.java:264)
Caused by: org.graalvm.compiler.graph.VerificationError: Object of type Lorg/graalvm/compiler/truffle/runtime/FrameWithoutBoxing; should not be materialized (must not pass virtual object into an invoke that cannot be inlined):
	at org.graalvm.compiler.nodes.virtual.EnsureVirtualizedNode.ensureVirtualFailure(EnsureVirtualizedNode.java:98)
	at org.graalvm.compiler.nodes.virtual.CommitAllocationNode.lower(CommitAllocationNode.java:117)
	at org.graalvm.compiler.phases.common.LoweringPhase$Round.process(LoweringPhase.java:465)
	at org.graalvm.compiler.phases.common.LoweringPhase$Round.access$200(LoweringPhase.java:341)
	at org.graalvm.compiler.phases.common.LoweringPhase$Round$ProcessFrame.preprocess(LoweringPhase.java:405)
	at org.graalvm.compiler.phases.common.LoweringPhase.processBlock(LoweringPhase.java:564)
	at org.graalvm.compiler.phases.common.LoweringPhase$Round.run(LoweringPhase.java:390)
	at org.graalvm.compiler.phases.Phase.run(Phase.java:49)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:197)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:139)
	at org.graalvm.compiler.phases.PhaseSuite.run(PhaseSuite.java:212)
	at org.graalvm.compiler.phases.common.IncrementalCanonicalizerPhase.run(IncrementalCanonicalizerPhase.java:56)
	at org.graalvm.compiler.phases.common.IncrementalCanonicalizerPhase.run(IncrementalCanonicalizerPhase.java:38)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:197)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:139)
	at org.graalvm.compiler.phases.common.LoweringPhase.lower(LoweringPhase.java:261)
	at org.graalvm.compiler.phases.common.LoweringPhase.run(LoweringPhase.java:254)
	at org.graalvm.compiler.phases.common.LoweringPhase.run(LoweringPhase.java:93)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:197)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:139)
	at org.graalvm.compiler.phases.PhaseSuite.run(PhaseSuite.java:212)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:197)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:139)
	at org.graalvm.compiler.core.GraalCompiler.emitFrontEnd(GraalCompiler.java:220)
	at org.graalvm.compiler.core.GraalCompiler.compile(GraalCompiler.java:144)
	at org.graalvm.compiler.core.GraalCompiler.compileGraph(GraalCompiler.java:129)
	at org.graalvm.compiler.truffle.compiler.TruffleCompilerImpl.compilePEGraph(TruffleCompilerImpl.java:518)
	at org.graalvm.compiler.truffle.compiler.TruffleCompilerImpl.compileAST(TruffleCompilerImpl.java:452)
	at org.graalvm.compiler.truffle.compiler.TruffleCompilerImpl$TruffleCompilationWrapper.performCompilation(TruffleCompilerImpl.java:621)
	at org.graalvm.compiler.truffle.compiler.TruffleCompilerImpl$TruffleCompilationWrapper.performCompilation(TruffleCompilerImpl.java:568)
	at org.graalvm.compiler.core.CompilationWrapper.run(CompilationWrapper.java:177)
	at org.graalvm.compiler.truffle.compiler.TruffleCompilerImpl.doCompile(TruffleCompilerImpl.java:280)
	at org.graalvm.compiler.truffle.runtime.GraalTruffleRuntime.doCompile(GraalTruffleRuntime.java:685)
	at org.graalvm.compiler.truffle.runtime.GraalTruffleRuntime.doCompile(GraalTruffleRuntime.java:709)
	at org.graalvm.compiler.truffle.runtime.BackgroundCompileQueue$Request.run(BackgroundCompileQueue.java:83)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at org.graalvm.compiler.truffle.runtime.BackgroundCompileQueue$TruffleCompilerThreadFactory$1.run(BackgroundCompileQueue.java:163)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:473)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)

.......................................................................................................................................................................................................................................................................................

Finished in 67.377574s, 28.0360 runs/s, 75.9600 assertions/s.
```

BTW: It's possible to improve the error where would be easy to locale in which Ruby script has happened this

**Comment by aardvark179** [comment]
I've not been able to reproduce this, either in jvm or native mode, so it may be something that has now been fixed. Can you try this on RC15 when its out and let us know if you still see the problem?

**Comment by eregon** [comment]
> BTW: It's possible to improve the error where would be easy to locale in which Ruby script has happened this

Unfortunately that is not easy, because this error happens on a separate thread for compilation, and the original thread is already executing different code with a different stack when this error happens during compilation. Taking a stacktrace when sending the compilation request would be one way, but it's quite expensive. Another way is to disable background compilation to debug this (`--vm.Dgraal.TruffleBackgroundCompilation=false`).

**Comment by deepj** [comment]
@aardvark179 I guess, this can be closed. It seems to be fine in RC15 and in the recent master.

**Comment by deepj** [comment]
In fact, it's reproducible with almost recent master which I have in `rom-sql` specs. Sequel 5.19.0 is used in this case

```
$ bundle exec rake spec
FFFF~/.rubies/truffleruby-1.0.0-nightly/bin/truffleruby -I~/.gem/truffleruby/2.6.2/gems/rspec-core-3.8.0/lib:~/.gem/truffleruby/2.6.2/gems/rspec-support-3.8.0/lib ~/.gem/truffleruby/2.6.2/gems/rspec-core-3.8.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb
FFFFFFFFF..~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb:53: warning: loading in progress, circular require considered harmful - ~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb
~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb:101: warning: loading in progress, circular require considered harmful - ~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb
~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb:136: warning: loading in progress, circular require considered harmful - ~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb
~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb:181: warning: loading in progress, circular require considered harmful - ~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb
.....FFF..............................FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF.FFF
Randomized with seed 6906
.FFFFFF.FFFFFFFFF.FFFFFFFFFFFFFFFFF.FFFFF.FFFFFFFFFFFF.FFFF.FF.FFFFFFFF...~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb:53: warning: loading in progress, circular require considered harmful - ~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb
~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb:101: warning: loading in progress, circular require considered harmful - ~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb
~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb:136: warning: loading in progress, circular require considered harmful - ~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb
~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb:181: warning: loading in progress, circular require considered harmful - ~/.gem/truffleruby/2.6.2/gems/rspec-expectations-3.8.2/lib/rspec/matchers/built_in/be.rb
...............FF.FFFFFFFFF.FFFFFFFF..[truffle] opt fail         Sequel::Postgres::Adapter#execute ~/.gem/truffleruby/2.6.2/gems/sequel-5.19.0/lib/sequel/adapters/postgres.rb:138|Reason org.graalvm.compiler.code.SourceStackTraceBailoutException$1: Object of type Lorg/graalvm/compiler/truffle/runtime/FrameWithoutBoxing; should not be materialized (must not let virtual object escape at node 2084|EndNode):
org.graalvm.compiler.code.SourceStackTraceBailoutException$1: Object of type Lorg/graalvm/compiler/truffle/runtime/FrameWithoutBoxing; should not be materialized (must not let virtual object escape at node 2084|EndNode):
	at org.truffleruby.language.control.FrameOnStackMarker.setNoLongerOnStack(FrameOnStackMarker.java:25)
	at org.truffleruby.language.control.FrameOnStackNode.execute(FrameOnStackNode.java:37)
	at org.truffleruby.language.methods.CatchBreakNode.execute(CatchBreakNode.java:38)
	at org.truffleruby.language.control.SequenceNode.execute(SequenceNode.java:34)
	at org.truffleruby.language.methods.CatchForMethodNode.execute(CatchForMethodNode.java:38)
	at org.truffleruby.language.methods.ExceptionTranslatingNode.execute(ExceptionTranslatingNode.java:51)
	at org.truffleruby.language.RubyRootNode.execute(RubyRootNode.java:54)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callProxy(OptimizedCallTarget.java:325)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callInlined(OptimizedCallTarget.java:265)
	at org.graalvm.compiler.truffle.runtime.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:63)
	at org.truffleruby.language.dispatch.CachedDispatchNode.call(CachedDispatchNode.java:190)
	at org.truffleruby.language.dispatch.CachedBoxedDispatchNode.executeDispatch(CachedBoxedDispatchNode.java:94)
	at org.truffleruby.language.dispatch.DispatchHeadNode.dispatch(DispatchHeadNode.java:43)
	at org.truffleruby.language.dispatch.RubyCallNode.executeWithArgumentsEvaluated(RubyCallNode.java:108)
	at org.truffleruby.language.dispatch.RubyCallNode.execute(RubyCallNode.java:99)
	at org.truffleruby.language.methods.ExceptionTranslatingNode.execute(ExceptionTranslatingNode.java:51)
	at org.truffleruby.language.control.SequenceNode.execute(SequenceNode.java:34)
	at org.truffleruby.language.methods.CatchForProcNode.execute(CatchForProcNode.java:36)
	at org.truffleruby.language.RubyRootNode.execute(RubyRootNode.java:54)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callProxy(OptimizedCallTarget.java:325)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callInlined(OptimizedCallTarget.java:265)
	at org.graalvm.compiler.truffle.runtime.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:63)
	at org.truffleruby.language.yield.CallBlockNode.callBlockCached(CallBlockNode.java:48)
	at org.truffleruby.language.yield.CallBlockNodeGen.executeCallBlock(CallBlockNodeGen.java:39)
	at org.truffleruby.language.yield.YieldNode.dispatch(YieldNode.java:27)
	at org.truffleruby.language.yield.YieldExpressionNode.execute(YieldExpressionNode.java:65)
	at org.truffleruby.language.methods.ExceptionTranslatingNode.execute(ExceptionTranslatingNode.java:51)
	at org.truffleruby.language.exceptions.TryNode.execute(TryNode.java:50)
	at org.truffleruby.language.exceptions.EnsureNode.execute(EnsureNode.java:38)
	at org.truffleruby.language.control.SequenceNode.execute(SequenceNode.java:34)
	at org.truffleruby.language.methods.CatchForMethodNode.execute(CatchForMethodNode.java:38)
	at org.truffleruby.language.methods.ExceptionTranslatingNode.execute(ExceptionTranslatingNode.java:51)
	at org.truffleruby.language.RubyRootNode.execute(RubyRootNode.java:54)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callProxy(OptimizedCallTarget.java:325)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callInlined(OptimizedCallTarget.java:265)
	at org.graalvm.compiler.truffle.runtime.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:63)
	at org.truffleruby.language.dispatch.CachedDispatchNode.call(CachedDispatchNode.java:190)
	at org.truffleruby.language.dispatch.CachedBoxedDispatchNode.executeDispatch(CachedBoxedDispatchNode.java:94)
	at org.truffleruby.language.dispatch.CachedBoxedDispatchNode.executeDispatch(CachedBoxedDispatchNode.java:84)
	at org.truffleruby.language.dispatch.CachedBoxedDispatchNode.executeDispatch(CachedBoxedDispatchNode.java:84)
	at org.truffleruby.language.dispatch.CachedBoxedDispatchNode.executeDispatch(CachedBoxedDispatchNode.java:84)
	at org.truffleruby.language.dispatch.DispatchHeadNode.dispatch(DispatchHeadNode.java:43)
	at org.truffleruby.language.dispatch.RubyCallNode.executeWithArgumentsEvaluated(RubyCallNode.java:108)
	at org.truffleruby.language.dispatch.RubyCallNode.execute(RubyCallNode.java:99)
	at org.truffleruby.language.control.FrameOnStackNode.execute(FrameOnStackNode.java:35)
	at org.truffleruby.language.methods.CatchBreakNode.execute(CatchBreakNode.java:38)
	at org.truffleruby.language.locals.WriteLocalVariableNode.execute(WriteLocalVariableNode.java:36)
	at org.truffleruby.language.RubyNode.doExecuteVoid(RubyNode.java:54)
	at org.truffleruby.language.control.SequenceNode.execute(SequenceNode.java:31)
	at org.truffleruby.language.methods.CatchForMethodNode.execute(CatchForMethodNode.java:38)
	at org.truffleruby.language.methods.ExceptionTranslatingNode.execute(ExceptionTranslatingNode.java:51)
	at org.truffleruby.language.RubyRootNode.execute(RubyRootNode.java:54)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callProxy(OptimizedCallTarget.java:325)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callRoot(OptimizedCallTarget.java:315)
Caused by: org.graalvm.compiler.graph.VerificationError: Object of type Lorg/graalvm/compiler/truffle/runtime/FrameWithoutBoxing; should not be materialized (must not let virtual object escape at node 2084|EndNode):
	at org.graalvm.compiler.nodes.virtual.EnsureVirtualizedNode.ensureVirtualFailure(EnsureVirtualizedNode.java:98)
	at org.graalvm.compiler.nodes.virtual.CommitAllocationNode.lower(CommitAllocationNode.java:117)
	at org.graalvm.compiler.phases.common.LoweringPhase$Round.process(LoweringPhase.java:476)
	at org.graalvm.compiler.phases.common.LoweringPhase$Round.access$200(LoweringPhase.java:352)
	at org.graalvm.compiler.phases.common.LoweringPhase$Round$ProcessFrame.preprocess(LoweringPhase.java:416)
	at org.graalvm.compiler.phases.common.LoweringPhase.processBlock(LoweringPhase.java:575)
	at org.graalvm.compiler.phases.common.LoweringPhase$Round.run(LoweringPhase.java:401)
	at org.graalvm.compiler.phases.Phase.run(Phase.java:49)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:197)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:139)
	at org.graalvm.compiler.phases.PhaseSuite.run(PhaseSuite.java:212)
	at org.graalvm.compiler.phases.common.IncrementalCanonicalizerPhase.run(IncrementalCanonicalizerPhase.java:56)
	at org.graalvm.compiler.phases.common.IncrementalCanonicalizerPhase.run(IncrementalCanonicalizerPhase.java:38)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:197)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:139)
	at org.graalvm.compiler.phases.common.LoweringPhase.lower(LoweringPhase.java:272)
	at org.graalvm.compiler.phases.common.LoweringPhase.run(LoweringPhase.java:265)
	at org.graalvm.compiler.phases.common.LoweringPhase.run(LoweringPhase.java:95)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:197)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:139)
	at org.graalvm.compiler.phases.PhaseSuite.run(PhaseSuite.java:212)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:197)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:139)
	at org.graalvm.compiler.core.GraalCompiler.emitFrontEnd(GraalCompiler.java:220)
	at org.graalvm.compiler.core.GraalCompiler.compile(GraalCompiler.java:144)
	at org.graalvm.compiler.core.GraalCompiler.compileGraph(GraalCompiler.java:129)
	at org.graalvm.compiler.truffle.compiler.TruffleCompilerImpl.compilePEGraph(TruffleCompilerImpl.java:518)
	at org.graalvm.compiler.truffle.compiler.TruffleCompilerImpl.compileAST(TruffleCompilerImpl.java:452)
	at org.graalvm.compiler.truffle.compiler.TruffleCompilerImpl$TruffleCompilationWrapper.performCompilation(TruffleCompilerImpl.java:621)
	at org.graalvm.compiler.truffle.compiler.TruffleCompilerImpl$TruffleCompilationWrapper.performCompilation(TruffleCompilerImpl.java:568)
	at org.graalvm.compiler.core.CompilationWrapper.run(CompilationWrapper.java:177)
	at org.graalvm.compiler.truffle.compiler.TruffleCompilerImpl.doCompile(TruffleCompilerImpl.java:280)
	at org.graalvm.compiler.truffle.runtime.GraalTruffleRuntime.doCompile(GraalTruffleRuntime.java:687)
	at org.graalvm.compiler.truffle.runtime.GraalTruffleRuntime.doCompile(GraalTruffleRuntime.java:711)
	at org.graalvm.compiler.truffle.runtime.BackgroundCompileQueue$Request.run(BackgroundCompileQueue.java:83)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at org.graalvm.compiler.truffle.runtime.BackgroundCompileQueue$TruffleCompilerThreadFactory$1.run(BackgroundCompileQueue.java:163)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:473)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)

.FFFFFFFFFFFFFFFFFFFF..FFFFFFFFFFFFFF....FFFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFFF.FF.F.FFFFFFF.FFFFFFFFFFFFFFFFFFFFFF.FFFFFFFFF.FFFFFFFFFFFFFFFFFFFF
```

**Comment by eregon** [comment]
I fixed this in 2a0f8392e04740373382a4f9772f781f09cb8c70.
